conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/bitcoin/BitcoinOperations.scala

```diff
@@ -0,0 +1,100 @@
+package tech.cryptonomic.conseil.indexer.bitcoin
+
+import cats.effect.{Concurrent, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import fs2.Stream
+import slick.jdbc.PostgresProfile.api._
+import slickeffect.Transactor
+
+import tech.cryptonomic.conseil.common.config.Platforms.BitcoinBatchFetchConfiguration
+import tech.cryptonomic.conseil.common.rpc.RpcClient
+import tech.cryptonomic.conseil.common.bitcoin.{BitcoinPersistence, Tables}
+import tech.cryptonomic.conseil.common.bitcoin.rpc.BitcoinClient
+import tech.cryptonomic.conseil.indexer.config.{Custom, Depth, Everything, Newest}
+
+/**
+  * Bitcoin operations for Lorre.
+  *
+  * @param bitcoinClient Bitcoin client instance
+  * @param persistence Bitcoin persistence instance
+  * @param tx [[slickeffect.Transactor]] to perform a Slick operations on the database
+  */
+class BitcoinOperations[F[_]: Concurrent](
+    bitcoinClient: BitcoinClient[F],
+    persistence: BitcoinPersistence[F],
+    tx: Transactor[F],
+    batchConf: BitcoinBatchFetchConfiguration
+) extends LazyLogging {
+
+  /**
+    * Start Lorre with mode defined with [[Depth]].
+    *
+    * @param depth Can be: Newest, Everything or Custom
+    */
+  def loadBlocks(depth: Depth): Stream[F, Unit] =
+    Stream
+      .eval(tx.transact(Tables.Blocks.sortBy(_.height).take(1).result)) // get last block height
```

Comment:
Could we extract this operation to separate method? That might improve visibility. It sounds like "heavy" method and I think it's better to extract this than to make hide it in the middle. Although it will do exactly the same, we could also test this out, while having in separate method (just to test this specific method).

Reply:
Done